app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java

```diff
@@ -5,32 +5,60 @@
 
 import com.google.firebase.messaging.RemoteMessage;
 
-import org.json.JSONObject;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.activity.OoniRunActivity;
 
-import java.util.Map;
+import ly.count.android.sdk.messaging.CountlyPush;
 
 public class FirebaseMessagingService extends com.google.firebase.messaging.FirebaseMessagingService {
 	private static final String TAG = "FCM";
 
 	@Override
 	public void onMessageReceived(RemoteMessage remoteMessage) {
-		// TODO(developer): Handle FCM messages here.
-		// Not getting messages here? See why this may be: https://goo.gl/39bRNJ
-		Log.d(TAG, "Message from: " + remoteMessage.getFrom());
-		Log.d(TAG, "Message data payload: " + remoteMessage.getData());
-		Map<String, String> params = remoteMessage.getData();
-		// Check if message contains a data payload.
-		if (remoteMessage.getData().size() > 0) {
-			try {
-				//TODO-FUTURE we can use click_action instead of type
-				JSONObject data = new JSONObject(params.toString());
-			} catch (Exception e) {
-				System.out.println("JSONException " + e);
-			}
+		super.onMessageReceived(remoteMessage);
+
+		Log.d("DemoFirebaseService", "got new message: " + remoteMessage.getData());
+
+		// decode message data and extract meaningful information from it: title, body, badge, etc.
+		CountlyPush.Message message = CountlyPush.decodeMessage(remoteMessage.getData());
+
+		if (message != null && message.has("typ")) {
```

Comment:
What is the different between `typ` used here and `type` used below for OONI Run?

Reply:
it's a countly thing https://support.count.ly/hc/en-us/articles/360037754031-Android-SDK#fcm-integration

Reply:
I suggest see if we can avoid integrating this "typ" thing to implement the functionality we want. If we need it, then I think we should better document what is this and why we need it.

Reply:
Please, address this comment by either removing the code or explaining why we need it?